### ISSUE TITLE
MATH: For ADD, SUB, NEG how is SA, MD and ND sub-task output structured

### ISSUE THREAD

**Issue by PhilipQuirke** [PhilipQuirke/verified_transformers#28]
Read https://github.com/PhilipQuirke/verified_transformers/blob/main/mixed_model.md. This follows on from Issue #29 

In the ins1_mix_d6_l3_h4_t40K model, the attention head P18L0H1 and H2 perform the SA, MD and ND sub-tasks depending on the input question type. That is, the output is poly-semantic. How is the output structured so that later nodes can select either SA output or the MD output or the ND output? Investigate

PCA the output of P18L0H0 for ADD, SUB, NEG questions. Do this for ADD, SUB, NEG separately and then for a batch of questions with all three question types. 
Compare the PCAs.
What can we say about how the output of this node pair (especially as it relates to later nodes selecting either SA output or the MD output or the ND output)?

Do the same results hold for each token position in P15 to P20?

(The Colab VerifiedArithmeticAnalyse.ipynb part 19B has a clause for the ins1_mix_d6_l3_h4_t40K model listing the interesting nodes for the ADD case in terms of their PCA results. It is an example of running PCA against groups of questions. But it runs against a single node rather than a node pair.)

Useful diagrams related to this are:
- https://github.com/PhilipQuirke/verified_transformers/blob/main/assets/ins1_mix_d6_l3_h4_t40K_s372001AlgorithmPurposePerNode.svg
- https://github.com/PhilipQuirke/verified_transformers/blob/main/assets/ins1_mix_d6_l3_h4_t40K_s372001QuantaAtP18.svg
- https://github.com/PhilipQuirke/verified_transformers/blob/main/assets/Hypothesis2_A2_Calc.png

**Comment by amirabdullah19852020** [comment]
Making a hard push on it this weekend - this is similar enough to Issue 29, that I think it'll be better to take both of these rather than one.

**Comment by amirabdullah19852020** [comment]
ADD, SUB and NEG are addition, subtraction(positive answer) and subtraction(negative answer)
SA: DN + DN' mod 10 (real valued operation)
MD: Dn- D′n%10, answer is positive
ND: Dn-D'n%10, answer is negative

**Comment by PhilipQuirke** [comment]
Yes, SA, MN, ND must each implement a map from 100 (10x10) input pairs Dn, D'n to 10 outputs states (0 to 9).

**Comment by amir-in-a-cynch** [comment]
What do you mean here by a "node pair"?

**Comment by amirabdullah19852020** [comment]
I've been able to clearout other responsibilities, and I intend to finish this in the coming fortnight for completeness.

**Comment by PhilipQuirke** [comment]
That's great news!
PQ
